File: THCS/2023_2024/TienGiang_2324/Bai1/ELCOST.py
def calculate_electricity_cost(kWh):
    if kWh <= 100:
        return kWh * 2500
    elif kWh <= 200:
        return 100 * 2500 + (kWh - 100) * 3000
    else:
        return 100 * 2500 + 100 * 3000 + (kWh - 200) * 3500

def main():
    # Đọc dữ liệu từ tệp
    with open("ELCOST.INP", "r") as input_file:
        n = int(input_file.readline().strip())  # Số lượng hộ gia đình
        kWh_list = list(map(int, input_file.readline().strip().split()))  # Số Kwh sử dụng

    total_cost = sum(calculate_electricity_cost(kWh) for kWh in kWh_list)  # Tính tổng chi phí

    # Ghi kết quả vào tệp
    with open("ELCOST.OUT", "w") as output_file:
        output_file.write(str(total_cost) + "\n")

if __name__ == "__main__":
    main()

File: THCS/2023_2024/TienGiang_2324/Bai1/ELCOST.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

int main() {
    ifstream input_file("ELCOST.INP");
    ofstream output_file("ELCOST.OUT");

    int n;
    input_file >> n; // Đọc số lượng hộ gia đình

    vector<int> kWh(n);
    for (int i = 0; i < n; ++i) {
        input_file >> kWh[i]; // Đọc số Kwh của mỗi hộ gia đình
    }

    long long total_cost = 0;

    for (int i = 0; i < n; ++i) {
        int usage = kWh[i];
        long long cost = 0;

        // Tính tiền điện cho hộ gia đình thứ i
        if (usage <= 100) {
            cost = usage * 2500;
        } else if (usage <= 200) {
            cost = 100 * 2500 + (usage - 100) * 3000;
        } else {
            cost = 100 * 2500 + 100 * 3000 + (usage - 200) * 3500;
        }

        total_cost += cost; // Cộng dồn vào tổng tiền
    }

    output_file << total_cost << endl; // Ghi tổng tiền vào tệp

    input_file.close();
    output_file.close();
    return 0;
}
